docs/source/reference/samplers/index.rst

```diff
@@ -11,6 +11,60 @@ The :mod:`~optuna.samplers` module defines a base class for parameter sampling a
 .. seealso::
     :ref:`user_defined_sampler` tutorial could be helpful if you want to implement your own sampler classes.
 
++----------------------------------+---------------+--------------+-------------+--------------+----------------------------------------+------------+----------------+
+|                                  | RandomSampler | GridSampler  | TPESampler  | CmaEsSampler |              NSGAIISampler             | QMCSampler | BoTorchSampler |
++==================================+===============+==============+=============+==============+========================================+============+================+
+| Float parameters                 |       ✅      |     ✅       |     ✅      |      ✅      |                   ▲                    |     ✅     |       ✅       |
++----------------------------------+---------------+--------------+-------------+--------------+----------------------------------------+------------+----------------+
+| Integer parameters               |       ✅      |     ✅       |     ✅      |      ✅      |                   ▲                    |     ✅     |       ✅       |
++----------------------------------+---------------+--------------+-------------+--------------+----------------------------------------+------------+----------------+
+| Categorical parameters           |       ✅      |     ✅       |     ✅      |      ▲       |                   ✅                   |     ▲      |       ✅       |
++----------------------------------+---------------+--------------+-------------+--------------+----------------------------------------+------------+----------------+
+| Pruning                          |       ▲       |     ▲        |     ✅      |      ▲       |                   ▲                    |     ▲      |       ▲        |
```

Comment:
RandomSampler, GridSampler and QMCSampler cannot consider pruned trials but it doesn't mean those algorithms will be insufficient.

```suggestion
| Pruning                          |       ✅       |     ✅        |     ✅      |      ▲       |                   ▲                    |     ✅      |       ▲        |
```

Regarding BoTorchSampler, I'm not sure how it will be insufficient. I'd like to hear the feedback from another reviewers.

Reply:
I think a check mark should be placed on this point in terms of whether the information from pruned trials can be used to improve sampling efficiency, since I think it is common to all methods that pruning terminates the trial midway and speeds up the sampling process except for multi-objective optimization. 

Therefore, I would like to use
- ▲  for RandomSampler, since it has no improvement in sample efficiency.
- ✅ for GridSampler, since it reduces the number of grids to consider.
- ✅ for QMCSampler, since it reduces the number of generated samples to consider.
- ❌ for NSGAIISampler, since the MO optimization has no interface for pruning.
- ▲  for BoTorchSampler, since it has no improvement in sample efficiency.

cc" @sile